### ISSUE TITLE
storage: The missing vdo helper message needs some patternflynization

### ISSUE THREAD

**Issue by KKoukiou** [cockpit-project/cockpit#18531]
### Explain what happens

![TestStoragePackagesVDO-testVdoMissingPackages-rhel-9-2-127 0 0 2-2801-FAIL](https://user-images.githubusercontent.com/14921356/226388214-9200ec53-bb92-437d-bb84-cd1e53e10370.png)

### Version of Cockpit

287

### Where is the problem in Cockpit?

Storage

### Server operating system

Fedora

### What browsers are you using?

Firefox

**Comment by garrett** [comment]
## Alert

So the error is used incorrectly. It should have a title and a body. It should probably keep linebreaks (but not be `pre`), else it will be hard to parse (as-is).

Here's a quick mockup example:

![Alert, inline, error details(1)](https://user-images.githubusercontent.com/10246/226412011-bbeaad16-7542-43b8-9633-e81bb0dff0ba.png)

We do have our own expand/contract errors, but PatternFly added them a couple years ago (after we had ours), called "expandable alerts": https://www.patternfly.org/v4/components/alert#expandable-alerts

I'm not sure if the > icon is enough. However, PF does have "View details" which may map to expanding? And I guess Ignore may map to the X? (I think it's to actually send a message to prevent it from coming back up later... but that doesn't matter in the context of our modal at least.)

## Help text

Additionally, there's too much spacing between the "Purpose" dropdown and the HelpText.

## Action button?

When something is going to be installed, that's an action. Should we change the text of the "Create" button change to "Install VDO and Create volume"? (It's wordy, but there's space.) It also makes it clear that something will be installed (in addition to the "Purpose" help text.

---

What do you think?

**Comment by omarY23** [comment]
Hey @garrett. Correct me if I am wrong but basically the tasks to be done are to make use of patternfly's expandeble alerts, change the icon of the alert and also reduce the padding between Purpose and helptext.

**Comment by omarY23** [comment]
Hey @garrett . I needed to ask a few things. In the following alert:

![226388214-9200ec53-bb92-437d-bb84-cd1e53e10370](https://user-images.githubusercontent.com/111054035/226967374-09666c4f-20db-4c07-a870-21af11f8b5c9.png)
If this error title, i.e. "Error creating volume" doesn't change for any case, should I hard code it as title for the alert and for the summary I can make use of index to get the rest part of the error string! For example:
initially -
`error_alert = <Alert variant='danger' isInline title={errors.toString()} />;`

My modification (according to patternfly's expandable alert component) -
```
error_alert = <Alert
                       isExpandable
                       isInline
                       variant="danger"
                       title="Error creating volume"                     // title of error
                       actionClose={<AlertActionCloseButton onClose={hideModalAlert} />}>
                <p>{errors_str.substr(23)}</p>                        //summary of error
             </Alert>
```
As the summary of this error starts from index 23

**Comment by garrett** [comment]
Oh, @KKoukiou is probably better to ask about that one. I do know that if we're using custom strings, they should be internationalized (in this case, translations). Otherwise, it would be better to try to use the error message when we can. Perhaps it can be split on the first colon (`:`)?

**Comment by omarY23** [comment]
> Oh, @KKoukiou is probably better to ask about that one. I do know that if we're using custom strings, they should be internationalized (in this case, translations).
> 
> Otherwise, it is usually better to try to use the actual error message whenever we can. Perhaps it can be split on the first colon (`:`)?

I can split the error message on the first colon to extract the title and separate the summary. That's also a way. Should I go ahead with that? @garrett @KKoukiou

**Comment by omarY23** [comment]
Hey @garrett. Can you tell me how to reproduce this issue. Especially the create logical volume dialog. 
My storage component has no options to create logical volume.
![Screenshot from 2023-03-31 01-05-28](https://user-images.githubusercontent.com/111054035/228946144-1c2bfbe9-2d6a-49bb-a58f-d5469a9c16fb.png)

**Comment by jelly** [comment]
For working on storage pages it's best to do this in a virtual machine so you can freely mess with storage without potentially losing any data by deleting / formatting your machines drives.

As vdo support is mostly a RHEL thing you'll need to prepare a rhel or centos VM with `./test/image-prepare centos-9-stream` and then `./bots/vm-run -S disk.img -s cockpit.socket`.

This adds a disk to the vm which you can play around with and requires you to create one with `qem-img` for example: `qemu-img create -f qcow2 disk.img 2G`.
